From 54c246ff1e5847434427a5035986d4fcf8daf3c8 Mon Sep 17 00:00:00 2001
From: Min He <min.he@intel.com>
Date: Fri, 4 Jan 2019 05:05:39 +0000
Subject: [PATCH 0942/1214] drm/i915: Not wait for fuse registers in guest

This patch will avoid waiting for fuse registers in GVT-g guest, to
reduce the booting time.

Change-Id: I05227b0e1f224b3b357320f56ee1895cd2e18737
Tracked-On: projectacrn/acrn-hypervisor/issues/2251
Signed-off-by: Min He <min.he@intel.com>
Reviewed-by: Zhao Yakui <yakui.zhao@intel.com>
Tracked-On: PKT-1648
---
 drivers/gpu/drm/i915/intel_runtime_pm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_runtime_pm.c b/drivers/gpu/drm/i915/intel_runtime_pm.c
index 6b5aa3b..c7cea89 100644
--- a/drivers/gpu/drm/i915/intel_runtime_pm.c
+++ b/drivers/gpu/drm/i915/intel_runtime_pm.c
@@ -387,7 +387,8 @@ static void hsw_power_well_enable(struct drm_i915_private *dev_priv,
 				  struct i915_power_well *power_well)
 {
 	enum i915_power_well_id id = power_well->id;
-	bool wait_fuses = power_well->hsw.has_fuses;
+	bool wait_fuses = power_well->hsw.has_fuses &&
+		!intel_vgpu_active(dev_priv);
 	enum skl_power_gate uninitialized_var(pg);
 	u32 val;
 
-- 
2.7.4

